packages/office-ui-fabric-react/src/components/TextField/TextField.base.tsx

```diff
@@ -174,48 +183,40 @@ export class TextField extends BaseComponent<ITextFieldProps, ITextFieldState> i
     const { isFocused } = this.state;
     const errorMessage = this._errorMessage;
 
+    this._classNames = getClassNames(styles!, {
+      theme: theme!,
```

Comment:
>! [](start = 18, length = 1)

Totally understand your usage here but FWIW, I took a different approach to this in Dropdown to minimize use of `!`

Reply:
@cliffkoh Maybe you could ... just describe the approach right here rather than making us look it up ;)

I will look in the other PR; i'm very curious how you could avoid this.

Reply:
I think @cliffkoh placed the theme prop as an "optional" in the styleprops definition so it pushes that issue down to styles.ts, but there he added a throw if not set. It's a bit harsh to have an optional that's gets thrown if not set :(

Reply:
I am in the middle of (hopefully) resolving this issue in createComponent...